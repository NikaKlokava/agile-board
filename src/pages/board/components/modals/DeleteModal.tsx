import { useDispatch, useSelector } from "react-redux";
import {
  deleteACtiveBoard,
  deleteBoard,
  deleteTask,
} from "../../../../redux/actionCreators/newBoardCreator";
import { Button } from "../../../../shared/components/button";
import { ModalWrapper } from "../../../../shared/components/modal_wrapper";
import cl from "./modal_styles.module.css";

type Props = {
  type: "board" | "task";
  activeName: string | undefined;
  taskUuid: string;
  onClose: () => void;
};

export const DeleteModal = ({ type, activeName, taskUuid, onClose }: Props) => {
  const activeBoard = useSelector<RootState, BoardType>(
    (state) => state.activeBoard
  );

  const dispatch = useDispatch();
  return (
    <ModalWrapper onWrapperClick={onClose}>
      <h2 className={cl.title} data-testid="delete-board">
        {`Delete this ${type}?`}
      </h2>
      <p className={cl.description}>
        {`Are you sure you want to delete the "${activeName}" ${type}? ${
          type === "board"
            ? "This action will remove all columns and tasks and cannot be reversed."
            : "This action will remove task and cannot be reversed."
        }`}
      </p>
      <div className={cl.btns_container}>
        <Button
          text={"Delete"}
          withIcon={false}
          testid={"delete-btn"}
          newClass="delete"
          type="button"
          onClick={() => {
            if (type === "board") {
              dispatch(deleteBoard(activeBoard.uuid));
              dispatch(deleteACtiveBoard());
            } else {
              dispatch(deleteTask(taskUuid));
            }
            onClose();
          }}
        />
        <Button
          text={"Cancel"}
          withIcon={false}
          testid={"delete-btn"}
          newClass="cancel"
          type="button"
          onClick={onClose}
        />
      </div>
    </ModalWrapper>
  );
};
