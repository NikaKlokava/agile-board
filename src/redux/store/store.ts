import { legacy_createStore } from "redux";
import { boardsReducer } from "../reducers/boardReducer";

const store = legacy_createStore(boardsReducer);

export default store;

// export type RootState = ReturnType<typeof store.getState>

// export interface RootState {
//     user: UserState;
//     loader: LoaderState;
//   }
// export type AppDispatch = typeof store.dispatch
