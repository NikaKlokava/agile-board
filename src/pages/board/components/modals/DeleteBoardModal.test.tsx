import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import store from "../../../../redux/store/store";
import { DeleteModal } from "./DeleteModal";

describe("Test the DeleteBoardModal component", () => {
  test("The DeleteBoardModal renders correctly", () => {
    const deleteBoardModalSnap = renderer
      .create(
        <Provider store={store}>
          <DeleteModal type={"board"} activeName={"test"} />
        </Provider>
      )
      .toJSON();
    expect(deleteBoardModalSnap).toMatchSnapshot();
  });
});
