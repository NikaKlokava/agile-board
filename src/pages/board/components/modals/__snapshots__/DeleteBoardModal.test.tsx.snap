// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the DeleteBoardModal component The DeleteBoardModal renders correctly 1`] = `
Array [
  <div
    className="modal_wrapper"
  />,
  <div
    className="modal_container"
  >
     
    <h2
      className="title"
      data-testid="delete-board"
    >
      Delete this board?
    </h2>
    <p
      className="description"
    >
      Are you sure you want to delete the "Platform Launch" board? This action will remove all columns and tasks and cannot be reversed.
    </p>
    <div
      className="btns_container"
    >
      <div
        className="button delete"
        data-testid="delete-btn"
      >
        <div
          className="button_text"
        >
          Delete
        </div>
      </div>
      <div
        className="button cancel"
        data-testid="delete-btn"
      >
        <div
          className="button_text"
        >
          Cancel
        </div>
      </div>
    </div>
  </div>,
]
`;
